politeiavoter/politeiavoter.go

```diff
@@ -615,6 +658,51 @@ func (c *ctx) _vote(token, voteId string) ([]string, *v1.BallotReply, error) {
 		return nil, nil, fmt.Errorf("no eligible tickets found")
 	}
 
+	// Put cast votes in a map so that we can
+	// filter in linear time
+	castVotes := make(map[string]v1.CastVote)
+	for _, v := range vrr.CastVotes {
+		castVotes[v.Ticket] = v
+	}
+
+	// Filter out tickets that have already voted
+	filtered := make([]*pb.CommittedTicketsResponse_TicketAddress, 0,
+		len(ctres.TicketAddresses))
+	for _, t := range ctres.TicketAddresses {
+		h, err := chainhash.NewHash(t.Ticket)
```

Comment:
Is there any reason this is being parsed to simply convert it right back to a string?

```Go
for _, t := range ctres.TicketAddresses {
	// Ticket has not voted yet.
	v, ok := castVotes[t.Ticket]
	if !ok {
		filtered = append(filtered, t)
		continue
	}

	// Move on to check validity....
}

Reply:
t.Ticket is a byte slice.  Hash.String() reverses the byte slice and converts it to hexadecimal.